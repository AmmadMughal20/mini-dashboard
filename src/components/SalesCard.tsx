
import React from 'react'
import { useSelector } from 'react-redux'
import { Card, CardContent, CardHeader, CardTitle } from '../components/ui/card'
import { AreaChart, Area, ResponsiveContainer } from 'recharts'
import type { RootState } from '../store'

/**
 * Sales dashboard card component displaying current sales and growth
 * Features a mini chart visualization showing sales trend
 */
const SalesCard: React.FC = () =>
{
  const { currentSales, growthPercentage, chartData } = useSelector((state: RootState) => state.sales)

  // Transform chart data for Recharts
  const chartDataFormatted = chartData.map((value: any, index: number) => ({
    name: `Day ${index + 1}`,
    sales: value
  }))


  return (
    <Card className="shadow-lg border-0 bg-white min-h-[375px] min-w-[320px]">
      <CardHeader className="pb-2">
        <CardTitle className="text-[34px] font-semibold text-slate-800">Sales</CardTitle>
      </CardHeader>
      <CardContent className="pt-0 flex flex-col justify-between h-full">
        <div className="flex flex-col space-y-3">
          <div className="items-baseline space-x-2">
            <span className="text-[54px] font-bold text-slate-900">
              ${currentSales.toLocaleString()}
            </span>
            <span className="text-[25px] font-medium text-green-600 flex items-center">
              <span className="mr-1">â†—</span>
              {growthPercentage}%
            </span>
          </div>
        </div>

        {/* Recharts area chart with gradient fill */}
        <div className="h-[150px] w-full">
          <ResponsiveContainer width="100%" height="100%">
            <AreaChart data={chartDataFormatted} margin={{ top: 5, right: 0, left: 0, bottom: 0 }}>
              <defs>
                <linearGradient id="salesGradient" x1="0" y1="0" x2="0" y2="1">
                  <stop offset="5%" stopColor="#669dee" stopOpacity={0.3} />
                  <stop offset="95%" stopColor="#669dee" stopOpacity={0.05} />
                </linearGradient>
              </defs>
              <Area
                type="monotone"
                dataKey="sales"
                stroke="#669dee"
                strokeWidth={3}
                fill="url(#salesGradient)"
                dot={false}
              />
            </AreaChart>
          </ResponsiveContainer>
        </div>
      </CardContent>
    </Card>
  )
}

export default SalesCard
