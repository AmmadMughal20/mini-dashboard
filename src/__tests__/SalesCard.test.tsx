import { render, screen } from '@testing-library/react'
import { Provider } from 'react-redux'
import { configureStore } from '@reduxjs/toolkit'
import SalesCard from '../components/SalesCard'
import salesReducer from '../features/salesSlice'

// Mock store for testing
const createMockStore = (initialState = {}) =>
{
  return configureStore({
    reducer: {
      sales: salesReducer,
    },
    preloadedState: {
      sales: {
        currentSales: 3200,
        growthPercentage: 12.5,
        chartData: [2800, 2900, 3000, 2950, 3100, 3200],
        ...initialState,
      },
    },
  })
}

describe('SalesCard', () =>
{
  it('renders sales data correctly', () =>
  {
    const store = createMockStore()

    render(
      <Provider store={store}>
        <SalesCard />
      </Provider>
    )

    expect(screen.getByText('Sales')).toBeInTheDocument()
    expect(screen.getByText('$3,200')).toBeInTheDocument()
    expect(screen.getByText('12.5%')).toBeInTheDocument()
  })

  it('matches snapshot', () =>
  {
    const store = createMockStore()

    const { container } = render(
      <Provider store={store}>
        <SalesCard />
      </Provider>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
