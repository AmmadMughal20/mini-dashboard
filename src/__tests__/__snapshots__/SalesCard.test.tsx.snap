// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SalesCard > matches snapshot 1`] = `
<div
  class="rounded-lg border bg-card text-card-foreground shadow-sm shadow-lg border-0 bg-white min-h-[375px] min-w-[320px]"
>
  <div
    class="flex flex-col space-y-1.5 p-6 pb-2"
  >
    <h3
      class="text-2xl font-semibold leading-none tracking-tight text-[34px] font-semibold text-slate-800"
    >
      Sales
    </h3>
  </div>
  <div
    class="p-6 pt-0 pt-0 flex flex-col justify-between h-full"
  >
    <div
      class="flex flex-col space-y-3"
    >
      <div
        class="items-baseline space-x-2"
      >
        <span
          class="text-[54px] font-bold text-slate-900"
        >
          $
          3,200
        </span>
        <span
          class="text-[25px] font-medium text-green-600 flex items-center"
        >
          <span
            class="mr-1"
          >
            â†—
          </span>
          12.5
          %
        </span>
      </div>
    </div>
    <div
      class="h-[150px] w-full"
    >
      <div
        class="recharts-responsive-container"
        style="width: 100%; height: 100%; min-width: 0;"
      />
    </div>
  </div>
</div>
`;
